src/colmap/scene/image.h

```diff
@@ -131,53 +131,92 @@ class Image {
   std::vector<struct Point2D> points2D_;
 };
 
+class Image : public BaseImage {
+ public:
+  explicit Image(struct Camera* camera);
+
+  // Initialized from an existing ``BaseImage`` with a raw camera pointer
+  Image(const BaseImage& base_image, struct Camera* camera);
+
+  // Access the address of the corresponding camera
+  inline struct Camera* Camera() const;
+
+  // Switch to different camera address
+  inline void SetCamera(struct Camera* camera);
+
+  // Disable ``SetCameraId``
+  void SetCameraId(camera_t camera_id) = delete;
+
+  // Reproject the 3D point onto the image in pixels (only when the camera
+  // pointer is available) Return false if the 3D point is behind the camera.
+  std::pair<bool, Eigen::Vector2d> ProjectPoint(
+      const Eigen::Vector3d& point3D) const;
+
+ private:
+  // The address of the corresponding camera
+  struct Camera* camera_;
```

Comment:
Hm, after looking at the code, I am not sure I would still go with the raw pointer but rather with the shared_ptr. There is no callsite where we would not use a shared_ptr for the camera?

Reply:
@ahojnnes How would you pass an element of `Reconstruction::cameras_` (unordered_map) as a `shared_ptr`?

@B1ueber2y is it legal to pass a NULLPTR camera? if not, 1) do you check for it? and 2) shouldn't `camera_` be a reference instead of a pointer?

Reply:
I agree that shared_ptr is not working well here. Since across COLMAP cameras are not initialized as shared pointers.

I didnt check for nullptr and could add it. And also, I agree that putting ``camera_`` as a reference is much better, and will make the updates. Thanks.